localstripe/resources.py

```diff
@@ -2878,3 +2878,180 @@ def __init__(self, card=None, customer=None, **kwargs):
+            # Fake status, because we cannot
+            # fake a cronable status update
+            if status is not None:
+                assert type(status) is str
+
+                if status not in ('paid', 'pending', 'failed'):
+                    raise UserError(400, 'Bad payout status')
+
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        # TODO
+        # Check that the amount, currency and source_type
+        # is avaliable via the Balance resource implemented in PR#165
+        # For the moment, we accept all payouts as of infinite balance
+        # and fake insufficient funds via the no_funds boolean parameter
+        if no_funds:
+            raise UserError(400, 'Insufficient Funds')
+
+        # All exceptions must be raised before this point.
+        super().__init__()
+
+        # Init defined variables
+        self.amount = amount
+        self.currency = currency
+        self.description = description
+        self.destination = destination
+        self.metadata = metadata
+        self.method = method
+        self.source_type = source_type
+        self.statement_descriptor = statement_descriptor
+
+        # By default expected arrival to two days after creation
+        two_days = 60 * 60 * 24 * 2
+        self.arrival_date = int(time.time() + two_days)
+
+        # As it was manually created
+        self.automatic = False
+        self.balance_transaction = f"txn_{random_id(24)}"
+
+        # Cancel related fields
+        self.failure_balance_transaction = None
+        self.failure_code = None
+        self.failure_message = None
+
+        # No payout reverse logic
+        self.original_payout = None
+        self.reversed_by = None
+
+        # Default status on creation
+        self.status = 'pending'
+
+        # Default
+        self.type = "bank_account"
+
+        schedule_webhook(Event('payout.created', self))
+
+        if status in ('paid', 'failed'):
+            self.status = status
+
+            # Define some arbitrary failure fields
+            # based on https://stripe.com/docs/api/payouts/failures
+            if status == 'failed':
+                self.failure_balance_transaction = ''
+                self.failure_code = 'could_not_process'
+                self.failure_message = \
+                    'The bank could not process this payout.'
+
+            schedule_webhook(Event(f'payout.{status}', self))
+
+    @classmethod
+    def _api_update(cls, id, **data):
+        payout = Payout._api_retrieve(id)
+        payout._update(**data)
+
+        schedule_webhook(Event('payout.updated', payout))
+
+        return payout
+
+    @classmethod
+    def _api_cancel(cls, id):
+        payout = Payout._api_retrieve(id)
+
+        # Only pending payouts can be canceled
+        if payout.status != 'pending':
+            raise UserError(400, 'Could not cancel not pending payout')
+
+        data = {"status": "canceled"}
+        payout._update(**data)
```

Comment:
This is equivalent to
```suggestion
        payout._update(status='canceled')
```